### DISCUSSION TITLE
write_solution_style=2

### DISCUSSION THREAD

**Discussion by pjaborges** [ERGO-Code/HiGHS#1051] [Q&A] [answered]
Hi!

I use highs as main solver in case my potential users do not have licenses for commercial solvers.
My console app calls the HiGHS executable after an LP file is generated.
Now, to bring the solution back to my system I was experiment the 3 different solutions styles ({0,1,2}) provided by HiGHS.
It was clear the most suitable format for me is style 2. However, 'sadly' is not printing the names of the variables and constraints.
Tough it follows some ordering extracted from the LP file. I cannot rely on that order to match variables and constraints to the architecture of my system since I use a lot of multithreading.

Any plan to enhance this file format to include the names of variables and constraints?

Thank you!

**Comment by jajhall** [comment, marked as answer]
v1.3 retains any constraint names read from a model file, and prints them in the solution file

**Reply by pjaborges** [reply]
> v1.3 retains any constraint names read from a model file, and prints them in the solution file

Yes, I do remember that this was solved for the standard output (when no specific style is used.).
I am now running 1.4.0 and when setting style = 2 names are not printed (see attachment).
![style_2_v1_4_0](https://user-images.githubusercontent.com/57194849/208959157-245bda0f-5e2a-44d1-83e1-479fe4f3d3f2.png)

**Reply by jajhall** [reply]
Now I've looked more carefully, I can't add names to `style = 2` since this is the raw solution style of Glpsol. 

I don't want to add another solution style to HiGHS. Can you not work with the HiGHS raw `style = 0`, since it gives the names.
